redskyctl/internal/commands/debug/metric.go

```diff
@@ -0,0 +1,313 @@
+	if err != nil {
+		return err
+	}
+
+	t.Status.StartTime = &startTime
+	t.Status.CompletionTime = &completionTime
+	return nil
+}
+
+func parseTime(input string, defaultTime time.Time) (metav1.Time, error) {
+	if input == "" {
+		return metav1.NewTime(defaultTime), nil
+	}
+
+	ss := regexp.MustCompile(`^([0-9]+)(?:\.([0-9]+))?$`).FindStringSubmatch(input)
+	if len(ss) == 3 {
+		sec, err := strconv.ParseInt(ss[1], 10, 64)
+		if err != nil {
+			return metav1.Time{}, err
+		}
+		nsec, err := strconv.ParseInt(ss[2], 10, 64)
+		if ss[2] != "" && err != nil {
+			return metav1.Time{}, err
+		}
+		return metav1.NewTime(time.Unix(sec, nsec)), nil
+	}
+
+	t, err := time.Parse(time.RFC3339Nano, input)
+	return metav1.NewTime(t), err
+}
+
+func parseDuration(input string, defaultDuration *metav1.Duration) (time.Duration, error) {
+	if input == "" {
+		if defaultDuration != nil {
+			return defaultDuration.Duration, nil
+		}
+		return 5 * time.Second, nil
+	}
+
+	return time.ParseDuration(input)
+}
+
+type promTest struct {
+	document *yaml.RNode
+}
+
+func newPromtest() *promTest {
+	return &promTest{
+		document: yaml.NewRNode(&yaml.Node{
+			Kind: yaml.DocumentNode,
+			Content: []*yaml.Node{
+				{
+					Kind: yaml.MappingNode,
+					Content: []*yaml.Node{
+						{Kind: yaml.ScalarNode, Value: "tests"},
+						{Kind: yaml.SequenceNode},
+					},
+				},
+			},
+		}),
+	}
+}
+
+func (p *promTest) writeTo(w kio.Writer) error {
+	return w.Write([]*yaml.RNode{p.document})
+}
+
+func (p *promTest) addHeadComment(pattern string, args ...interface{}) {
+	prefix := p.document.YNode().HeadComment
+	if prefix != "" {
+		prefix += "\n"
+	}
+
+	p.document.YNode().HeadComment = prefix + fmt.Sprintf(pattern, args...) + "\n"
+}
+
+func (p *promTest) addTest(t *redskyv1beta1.Trial) error {
+	return p.document.PipeE(
+		yaml.Lookup("tests"),
+		yaml.Append(&yaml.Node{
+			Kind: yaml.MappingNode,
+			Content: []*yaml.Node{
+				{Kind: yaml.ScalarNode, Value: "name"},
+				{Kind: yaml.ScalarNode, Value: t.Name},
+
+				{Kind: yaml.ScalarNode, Value: "interval", HeadComment: "\n"},
+				{Kind: yaml.ScalarNode, Value: "5s", LineComment: "Scrape interval"},
+
+				{Kind: yaml.ScalarNode, Value: "input_series"},
+				{
+					Kind: yaml.SequenceNode,
+					Content: []*yaml.Node{
+						{
+							Kind: yaml.MappingNode,
+							Content: []*yaml.Node{
+								{Kind: yaml.ScalarNode, Value: "series"},
+								{Kind: yaml.ScalarNode, Value: "", Style: yaml.SingleQuotedStyle},
+
+								{Kind: yaml.ScalarNode, Value: "values"},
+								{Kind: yaml.ScalarNode, Value: "", Style: yaml.SingleQuotedStyle},
+							},
+						},
+					},
+				},
+
+				{Kind: yaml.ScalarNode, Value: "promql_expr_test", HeadComment: "\n"},
+				{Kind: yaml.SequenceNode},
+			},
+		}),
+	)
+}
+
+func (p *promTest) addPromqlExprTest(t *redskyv1beta1.Trial, m *redskyv1beta1.Metric, q string) error {
+	return p.document.PipeE(
```

Comment:
I think this is good. 
We may want to do some bogus/fake data insertion which we could cheese by using the arithmetic-like expression ( I don't know the actual format, but im thinking something like `1x<duration in seconds>` so we could have a hopefully easily calculable value to validate against. 
 
```
  - input_series:
      - series: 'go_goroutines{job="prometheus", instance="localhost:9090"}'
        values: '10+10x2 30+20x5' # 10 20 30 30 50 70 90 110 130
      - series: 'go_goroutines{job="node_exporter", instance="localhost:9100"}'
        values: '10+10x7 10+30x4' # 10 20 30 40 50 60 70 80 10 40 70 100 130
```